Extension/src/main.ts

```diff
@@ -452,20 +455,57 @@ function rewriteManifest(installBlob: InstallBlob): Promise<void> {
         "onCommand:C_Cpp.TakeSurvey",
         "onDebug"
     ];
+}
 
-    // Remove the entry for cppdbg's proxy stub and replace it with the real debugger binary
-    util.packageJson.contributes.debuggers[0].runtime = undefined;
-    util.packageJson.contributes.debuggers[0].program = './debugAdapters/OpenDebugAD7';
-    util.packageJson.contributes.debuggers[0].windows = { "program": "./debugAdapters/bin/OpenDebugAD7.exe" };
-
-    // Remove the entry for cppvsdbg's proxy stub and replace it with the real debugger binary for Windows only.
-    if (os.platform() === 'win32') {
-        util.packageJson.contributes.debuggers[1].runtime = undefined;
-        util.packageJson.contributes.debuggers[1].program = './debugAdapters/vsdbg/bin/vsdbg.exe';
-    }
+// Registers adapterExecutableCommands for cppdbg and cppvsdbg. If it is not ready, it will prompt waiting for the download.
+// 
+// Note: util.extensionContext.extensionPath is needed for the commands because VsCode does not support relative paths for adapterExecutableComand
+function registerAdapterExecutableCommands(): void {
+    vscode.commands.registerCommand('extension.cppdbgAdapterExecutableCommand', () => {
+        return util.checkInstallLockFile().then(ready => {
+            if (ready)
+            {
+                let command: string = path.join(util.extensionContext.extensionPath, '/debugAdapters/OpenDebugAD7');
+
+                // Windows has the exe in debugAdapters/bin.
+                if (os.platform() === 'win32')
+                {
+                    command = path.join(util.extensionContext.extensionPath, "./debugAdapters/bin/OpenDebugAD7.exe");
+                }
 
-    if (util.packageJson.extensionFolderPath.includes(".vscode-insiders"))
-        util.packageJson.contributes.configuration.properties["C_Cpp.intelliSenseEngine"].default = "Default";
+                return {
+                    command: command
+                }
+            }
+            else {
+                util.showReloadOrWaitPromptOnce();
```

Comment:
Are you supposed to throw an error instead of returning null then?

Reply:
Tested in C# extension and did not work, they are currently adding a fix for this and will tell use how to properly handle errors.